### PR TITLE
Dah 2158 url filtering

### PR DESCRIPTION
**Pull request by chadbrokaw**
[DAH-2158](https://sfgovdt.jira.com/browse/DAH-2158)

[DAH-2158]: https://sfgovdt.jira.com/browse/DAH-2158?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by jimlin-sfgov** [comment]
note: I'm really disliking this limitation of the code climate config:
> Exclusions can only be made at the global level

It would have been nice to exclude React functional components from the Cognitive Complexity check.

**Comment by chadbrokaw** [comment]
> note: I'm really disliking this limitation of the code climate config:
> 
> > Exclusions can only be made at the global level
> 
> It would have been nice to exclude React functional components from the Cognitive Complexity check.

Agreed!

File: app/javascript/components/lease_ups/LeaseUpApplicationsFilterContainer.js [hunk 1]
```diff
@@ -60,6 +62,32 @@ const LeaseUpApplicationsFilterContainer = ({
...
+    const filterFormState = form.getState()
+    const filterFormKeys = Object.keys(filterFormState.values)
+    const newURLSearchParams = new URLSearchParams()
+
+    if (!filterFormKeys || filterFormKeys.length === 0) {
+      setSearchParams({})
+      return
+    }
+
+    filterFormKeys.forEach((filterKey) => {
```
Comment (jimlin-sfgov) [inline app/javascript/components/lease_ups/LeaseUpApplicationsFilterContainer.js]: question: was there any reason not to use `for...in` iteration style?

**Review by jimlin-sfgov — APPROVED** [review]
great work, liking the tests as well!

**Pull request merged**